packages/backend/src/api/controllers/ForcedActionsController.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  renderNewPerpetualForcedTradePage,
+  renderNewSpotForcedWithdrawPage,
+} from '@explorer/frontend'
+import { UserDetails } from '@explorer/shared'
+import { AssetHash, AssetId, EthereumAddress } from '@explorer/types'
+
+import { UserService } from '../../core/UserService'
+import { AssetRepository } from '../../peripherals/database/AssetRepository'
+import { PreprocessedAssetHistoryRepository } from '../../peripherals/database/PreprocessedAssetHistoryRepository'
+import { ControllerResult } from './ControllerResult'
+
+export class ForcedActionsController {
+  constructor(
+    private readonly userService: UserService,
+    private readonly preprocessedAssetHistoryRepository: PreprocessedAssetHistoryRepository<
+      AssetHash | AssetId
+    >,
+    private readonly assetRepository: AssetRepository
+  ) {}
+
+  async getSpotForcedWithdrawalPage(
+    givenUser: Partial<UserDetails>,
+    vaultId: bigint
+  ): Promise<ControllerResult> {
+    const user = await this.userService.getUserDetails(givenUser)
+
+    if (!user) {
+      return { type: 'not found', content: 'User must be logged in' }
```

Comment:
Maybe we should handle those cases with some kind of error page like the 404 one? Definitely not a discussion for this PR but I think it would be a good UX addition